Libplanet/Net/Protocols/RoutingTable.cs

```diff
@@ -246,6 +232,22 @@ public void Check(BoundPeer peer, DateTimeOffset start, DateTimeOffset end)
             BucketOf(peer).Check(peer, start, end);
         }
 
+        internal void AddPeer(BoundPeer peer, DateTimeOffset updated)
+        {
+            if (peer is null)
+            {
+                throw new ArgumentNullException(nameof(peer));
+            }
+
+            if (peer.Address.Equals(_address))
+            {
+                throw new ArgumentException("Cannot add self to routing table.");
```

Comment:
This piece of code is not recently written, but…

```suggestion
                throw new ArgumentException(
                    "A peer is disallowed to add itself to its routing table.",
                    nameof(peer)
                );
```